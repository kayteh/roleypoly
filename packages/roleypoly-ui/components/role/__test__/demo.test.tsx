/**
 *  @jest-environment jsdom
 */
/* eslint-env jest */
import * as React from 'react'
import { shallow } from 'enzyme'
import RoleDemo from '../demo'
import 'jest-styled-components'

describe('<RoleDemo />', () => {
  it('renders', () => {
    const demo = shallow(<RoleDemo role={{ name: 'test demo role', color: '#ffffff' }} />)
    expect(demo).toMatchSnapshot()
  })

  it('changes state when clicked', () => {
    const el = <RoleDemo role={{ name: 'test demo role', color: '#ffffff' }} />
    const demo = shallow<typeof el>(el)
    expect(demo.state().active).toEqual(false)
    demo.dive().simulate('click')
    expect(demo.state().active).toEqual(true)
  })
})
