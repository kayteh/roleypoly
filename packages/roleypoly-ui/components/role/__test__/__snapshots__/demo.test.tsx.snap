// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoleDemo /> renders 1`] = `
<Role
  active={false}
  onToggle={[Function]}
  role={
    Object {
      "color": "#ffffff",
      "name": "test demo role",
    }
  }
/>
`;
